models/Element/Tag.php

```diff
@@ -338,4 +369,30 @@ public function correctPath()
             $this->idPath = '/';
         }
     }
+
+    /**
+     * Deletes a tag
+     *
+     * @throws \Exception
+     */
+    public function delete()
+    {
+        try {
+            \Pimcore::getEventDispatcher()->dispatch(TagEvents::PRE_DELETE, new TagEvent($this));
+
+            $this->getDao()->delete();
+
+            \Pimcore::getEventDispatcher()->dispatch(TagEvents::POST_DELETE, new TagEvent($this));
+        } catch (\Exception $e) {
+            throw $e;
+        }
```

Comment:
The try-catch is unneeded if it only re-throws the exception.

```suggestion
        \Pimcore::getEventDispatcher()->dispatch(TagEvents::PRE_DELETE, new TagEvent($this));

        $this->getDao()->delete();

        \Pimcore::getEventDispatcher()->dispatch(TagEvents::POST_DELETE, new TagEvent($this));
```

Reply:
yes, valid point. thanks!